src/commands.ts

```diff
@@ -1132,89 +1004,105 @@ const getParentNode = (selection: Selection) => {
 }
 
 // TODO:: remove this check when we allow all type of block node to have comment
-const isAllowedType = (type: NodeType) =>
+const isCommentingAllowed = (type: NodeType) =>
   type === schema.nodes.section ||
+  type === schema.nodes.citation ||
+  type === schema.nodes.bibliography_item ||
   type === schema.nodes.footnotes_section ||
   type === schema.nodes.bibliography_section ||
   type === schema.nodes.keyword_group ||
   type === schema.nodes.paragraph ||
   type === schema.nodes.figure_element ||
   type === schema.nodes.table_element
 
-const getNode = (node: ManuscriptNode) => {
-  if (node.type === schema.nodes.keywords) {
-    const keywordGroups = findChildrenByType(
-      node,
-      schema.nodes.keyword_group,
-      true
-    )
-    return keywordGroups.length ? keywordGroups[0].node : node
-  }
-  return node
-}
-
-const addBlockComment = (
+export const addNodeComment = (
   node: ManuscriptNode,
   state: ManuscriptEditorState,
   dispatch?: Dispatch
 ) => {
-  const {
-    attrs: { id },
-    type,
-  } = getNode(node)
-  const comment = buildComment(id)
-  const tr = state.tr.setMeta(commentAnnotation, {
-    [SET_COMMENT]: comment,
-  })
-
-  tr.setMeta('addToHistory', false)
-
-  if (isAllowedType(type)) {
-    dispatch && dispatch(tr)
-    return true
-  } else {
+  if (!isCommentingAllowed(node.type)) {
```

Comment:
we need to allow user to add a block comment to sections, that will be in case user create a comment from three-dot button when selection is no section title

Reply:
Adding comments to sections should be allowed, and the mapping from section titles to sections happens in `context-menu.ts`.

Reply:
ok go it, but we need to return that parent node https://github.com/Atypon-OpenSource/manuscripts-body-editor/pull/291/files#diff-c391703ced166af395b8380f6eef0056b5f0aab33e2f80ccf1e10ce44e9af80aR601

Reply:
Right, I'll fix that.